src/javascript/request-composer/request.js

```diff
@@ -0,0 +1,318 @@
+// Copyright 2016 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import moment from 'moment';
+
+function buildReportRequest(params) {
```

Comment:
Can you add [JS doc](http://usejsdoc.org/) comments to all the functions/methods in this PR?

I've been a bit lax about it in some of my existing code, but I think we should standardize on enforcing it. It'll definitely make code reviews easier.

I'll add it to the linter and make sure all the other code in this repo passes.